ui2.0/app/sagas/index.js

```diff
@@ -130,43 +147,18 @@ function* updateStageSaga({
       if (nextStage === 'login_input') {
         yield put(showIncorrect());
       }
-      if (nextStage === 'complete') {
-        yield put(showCorrect());
-      }
 
       if (transitionDelay < 1) {
-        delay(transitionDelay * 1000);
+        yield delay(transitionDelay * 1000);
         yield put(hideIncorrect());
-        yield put(hideCorrect());
       }
 
-      // if (nextStage === 'login_input') {
-      //   const { remainingTrials, nextPenaltyTime } = params;
-      //   const penaltyTime = transitionDelay ? transitionDelay : nextPenaltyTime;
-      //   yield put(
-      //     setIncorrectParam({
-      //       remainingTrials,
-      //       nextPenaltyTime: penaltyTime
-      //     })
-      //   );
-
-      //   yield put(showIncorrect());
-
-      // if (transitionDelay !== 0) {
-      //   const { remainingTrials, nextPenaltyTime } = params;
-      //   yield put(setIncorrectParam({ remainingTrials, nextPenaltyTime }));
-      //   yield put(showIncorrect());
-      //   if (transitionDelay < 1) {
-      //     yield delay(transitionDelay * 1000);
-      //     yield put(hideIncorrect());
-      //   }
-      // }
-      // } else if (transitionDelay !== 0) {
-      //   yield put(showCorrect());
-      //   yield delay(transitionDelay * 1000);
-      //   yield put(hideCorrect());
-      //   yield put(gotoComplete());
-      // }
+      if (nextStage === 'complete') {
```

Comment:
this case should be checked before `transitionDelay < 1` and remove the delay, so that it won't be delayed twice